src/serac/infrastructure/logger.cpp

```diff
@@ -35,43 +40,84 @@ bool initialize(MPI_Comm comm)
     return false;
   }
 
-  // Separate streams for different message levels
-  slic::LogStream* iStream  = nullptr;  // info
-  slic::LogStream* dStream  = nullptr;  // debug
-  slic::LogStream* weStream = nullptr;  // warnings and errors
+  // Console streams, std::cout for info/debug, std::cerr for warnings/errors
+  slic::LogStream* i_console_logstream  = nullptr;  // info
+  slic::LogStream* d_console_logstream  = nullptr;  // debug
+  slic::LogStream* we_console_logstream = nullptr;  // warnings and errors
+
+  // File streams, all message levels go to one file
+  logger_ofstream = std::make_unique<std::ofstream>();
+  if (rank == 0) {
+    // Only root node writes/opens the file, other nodes will have a noop stream
+    logger_ofstream = std::make_unique<std::ofstream>();
```

Comment:
Just seems like this is created one extra time for rank 0? Not sure if it's needed or not.

Reply:
... don't know what you are talking about... this line never existed...